src/util/prolog/prologResult.ts

```diff
@@ -0,0 +1,54 @@
+export class PrologResult {
+	private readonly results: string[][];
+	private readonly rawResults: string[];
+
+	constructor(rawResults: string[]) {
+		this.rawResults = rawResults;
+		this.results = [];
+	}
+
+	public getResults(): string[][] {
+		if (this.results.length >= 1 ) { return this.results; }
+
+		this.rawResults.forEach((e: any) => {
+			this.results.push(this.rawResultsToInnerArray(e));
+		});
+
+		return this.results;
+	}
+
+	public filterBooleanResults(): string[][] {
+		const copy: string[][] = this.getResults();
+		return copy.filter((e: any) => {
+			return e[0] === "true" || e[0] === "false";
```

Comment:
what about checking the length of `e`? Also, you can write this as `.filter((e: any) => e[0] === "true" ...)`

Reply:
Will be solved by utilizing `RegExp` groups now in another function.